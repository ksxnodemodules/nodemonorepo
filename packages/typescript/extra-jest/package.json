{
  "name": "extra-jest",
  "version": "0.0.3",
  "main": "index.js",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/ksxnodemodules/nodemonorepo#readme",
  "bugs": {
    "url": "https://github.com/ksxnodemodules/nodemonorepo/issues"
  },
  "dependencies": {
    "ramda": "^0.25.0",
    "fs-extra": "^6.0.0",
    "fs-tree-utils": "^0.0.3",
    "unique-temp-path": "^0.0.3",
    "@types/ramda": "^0.25.24",
    "@types/fs-extra": "^5.0.1",
    "@types/node": "^10.0.0"
  },
  "peerDependencies": {
    "jest": "^22.4.3"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
