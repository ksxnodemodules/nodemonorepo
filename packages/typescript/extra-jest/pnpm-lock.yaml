lockfileVersion: 5.3

specifiers:
  '@types/fs-extra': ^8.0.1
  '@types/js-yaml': ^3.12.1
  '@types/node': ^14.0.0
  fs-extra: ^8.1.0
  fs-tree-utils: ^0.1.8
  js-yaml: ^3.13.1
  tslib: ^1.10.0
  unique-temp-path: ^0.1.3

dependencies:
  '@types/fs-extra': 8.0.1
  '@types/js-yaml': 3.12.1
  '@types/node': 14.17.21
  fs-extra: 8.1.0
  fs-tree-utils: link:../fs-tree-utils
  js-yaml: 3.13.1
  tslib: 1.10.0
  unique-temp-path: link:../unique-temp-path

packages:

  /@types/fs-extra/8.0.1:
    resolution: {integrity: sha512-J00cVDALmi/hJOYsunyT52Hva5TnJeKP5yd1r+mH/ZU0mbYZflR0Z5kw5kITtKTRYMhm1JMClOFYdHnQszEvqw==}
    dependencies:
      '@types/node': 13.1.6
    dev: false

  /@types/js-yaml/3.12.1:
    resolution: {integrity: sha512-SGGAhXLHDx+PK4YLNcNGa6goPf9XRWQNAUUbffkwVGGXIxmDKWyGGL4inzq2sPmExu431Ekb9aEMn9BkPqEYFA==}
    dev: false

  /@types/node/13.1.6:
    resolution: {integrity: sha512-Jg1F+bmxcpENHP23sVKkNuU3uaxPnsBMW0cLjleiikFKomJQbsn0Cqk2yDvQArqzZN6ABfBkZ0To7pQ8sLdWDg==}
    dev: false

  /@types/node/14.17.21:
    resolution: {integrity: sha512-zv8ukKci1mrILYiQOwGSV4FpkZhyxQtuFWGya2GujWg+zVAeRQ4qbaMmWp9vb9889CFA8JECH7lkwCL6Ygg8kA==}
    dev: false

  /argparse/1.0.10:
    resolution: {integrity: sha512-o5Roy6tNG4SL/FOkCAN6RzjiakZS25RLYFrcMttJqbdd8BWrnA+fGz57iN5Pb06pvBGvl5gQ0B48dJlslXvoTg==}
    dependencies:
      sprintf-js: 1.0.3
    dev: false

  /esprima/4.0.1:
    resolution: {integrity: sha512-eGuFFw7Upda+g4p+QHvnW0RyTX/SVeJBDM/gCtMARO0cLuT2HcEKnTPvhjV6aGeqrCB/sbNop0Kszm0jsaWU4A==}
    engines: {node: '>=4'}
    hasBin: true
    dev: false

  /fs-extra/8.1.0:
    resolution: {integrity: sha512-yhlQgA6mnOJUKOsRUFsgJdQCvkKhcz8tlZG5HBQfReYZy46OwLcY+Zia0mtdHsOo9y/hP+CxMN0TU9QxoOtG4g==}
    engines: {node: '>=6 <7 || >=8'}
    dependencies:
      graceful-fs: 4.2.2
      jsonfile: 4.0.0
      universalify: 0.1.2
    dev: false

  /graceful-fs/4.2.2:
    resolution: {integrity: sha512-IItsdsea19BoLC7ELy13q1iJFNmd7ofZH5+X/pJr90/nRoPEX0DJo1dHDbgtYWOhJhcCgMDTOw84RZ72q6lB+Q==}
    dev: false

  /js-yaml/3.13.1:
    resolution: {integrity: sha512-YfbcO7jXDdyj0DGxYVSlSeQNHbD7XPWvrVWeVUujrQEoZzWJIRrCPoyk6kL6IAjAG2IolMK4T0hNUe0HOUs5Jw==}
    hasBin: true
    dependencies:
      argparse: 1.0.10
      esprima: 4.0.1
    dev: false

  /jsonfile/4.0.0:
    resolution: {integrity: sha1-h3Gq4HmbZAdrdmQPygWPnBDjPss=}
    optionalDependencies:
      graceful-fs: 4.2.2
    dev: false

  /sprintf-js/1.0.3:
    resolution: {integrity: sha1-BOaSb2YolTVPPdAVIDYzuFcpfiw=}
    dev: false

  /tslib/1.10.0:
    resolution: {integrity: sha512-qOebF53frne81cf0S9B41ByenJ3/IuH8yJKngAX35CmiZySA0khhkovshKK+jGCaMnVomla7gVlIcc3EvKPbTQ==}
    dev: false

  /universalify/0.1.2:
    resolution: {integrity: sha512-rBJeI5CXAlmy1pV+617WB9J63U6XcazHHF2f2dbJix4XzpUF0RS3Zbj0FGIOCAva5P/d/GBOYaACQ1w+0azUkg==}
    engines: {node: '>= 4.0.0'}
    dev: false
