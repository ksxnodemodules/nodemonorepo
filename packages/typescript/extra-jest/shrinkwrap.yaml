dependencies:
  '@types/fs-extra': 5.0.2
  '@types/node': 10.0.1
  '@types/ramda': 0.25.24
  fs-extra: 6.0.0
  fs-tree-utils: 'link:../fs-tree-utils'
  ramda: 0.25.0
  unique-temp-path: 'link:../unique-temp-path'
packages:
  /@types/fs-extra/5.0.2:
    dependencies:
      '@types/node': 10.0.1
    dev: false
    resolution:
      integrity: sha512-Q3FWsbdmkQd1ib11A4XNWQvRD//5KpPoGawA8aB2DR7pWKoW9XQv3+dGxD/Z1eVFze23Okdo27ZQytVFlweKvQ==
  /@types/node/10.0.1:
    dev: false
    resolution:
      integrity: sha512-HKTs07TC1RJRBd9z0Kf/cy37j2HCbUfcVKS6Xm9orZVcLqZ1JXWe/JTDLw6AwxmLOxV45fDpOMPXyqXU3TfOPQ==
  /@types/ramda/0.25.24:
    dev: false
    resolution:
      integrity: sha512-c0TmWA7d4y9KLJJwL/cLPEfSReSgFQK9BtemcCATT48lMeyD7HG8IfGY8bamSuz/Byx1l+13hZV0PCvHsgMB3w==
  /fs-extra/6.0.0:
    dependencies:
      graceful-fs: 4.1.11
      jsonfile: 4.0.0
      universalify: 0.1.1
    dev: false
    resolution:
      integrity: sha512-lk2cUCo8QzbiEWEbt7Cw3m27WMiRG321xsssbcIpfMhpRjrlC08WBOVQqj1/nQYYNnPtyIhP1oqLO3QwT2tPCw==
  /graceful-fs/4.1.11:
    dev: false
    engines:
      node: '>=0.4.0'
    resolution:
      integrity: sha1-Dovf5NHduIVNZOBOp8AOKgJuVlg=
  /jsonfile/4.0.0:
    dev: false
    optionalDependencies:
      graceful-fs: 4.1.11
    resolution:
      integrity: sha1-h3Gq4HmbZAdrdmQPygWPnBDjPss=
  /ramda/0.25.0:
    dev: false
    resolution:
      integrity: sha512-GXpfrYVPwx3K7RQ6aYT8KPS8XViSXUVJT1ONhoKPE9VAleW42YE+U+8VEyGWt41EnEQW7gwecYJriTI0pKoecQ==
  /universalify/0.1.1:
    dev: false
    resolution:
      integrity: sha1-+nG63UQ3r0wUiEHjs7Fl+enlkLc=
registry: 'https://registry.npmjs.org/'
shrinkwrapMinorVersion: 5
shrinkwrapVersion: 3
specifiers:
  '@types/fs-extra': ^5.0.1
  '@types/node': ^10.0.0
  '@types/ramda': ^0.25.24
  fs-extra: ^6.0.0
  fs-tree-utils: ^0.0.3
  ramda: ^0.25.0
  unique-temp-path: ^0.0.3
