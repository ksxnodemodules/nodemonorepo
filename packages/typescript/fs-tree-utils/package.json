{
  "name": "fs-tree-utils",
  "version": "0.0.3",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/ksxnodemodules/nodemonorepo#readme",
  "bugs": {
    "url": "https://github.com/ksxnodemodules/nodemonorepo/issues"
  },
  "main": "index.js",
  "dependencies": {
    "fs-extra": "^6.0.0",
    "ramda": "^0.25.0",
    "@types/fs-extra": "^5.0.1",
    "@types/ramda": "^0.25.21",
    "@types/node": "^10.0.0"
  },
  "devDependencies": {
    "unique-temp-path": "*"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
