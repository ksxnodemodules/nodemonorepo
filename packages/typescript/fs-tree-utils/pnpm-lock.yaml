dependencies:
  '@types/fs-extra': 8.0.0
  '@types/node': 11.13.17
  '@types/ramda': 0.26.15
  fs-extra: 8.1.0
  monorepo-shared-assets: 'link:../monorepo-shared-assets'
  ramda: 0.26.1
  tslib: 1.10.0
devDependencies:
  unique-temp-path: 'link:../unique-temp-path'
lockfileVersion: 5.1
packages:
  /@types/fs-extra/8.0.0:
    dependencies:
      '@types/node': 11.13.17
    dev: false
    resolution:
      integrity: sha512-bCtL5v9zdbQW86yexOlXWTEGvLNqWxMFyi7gQA7Gcthbezr2cPSOb8SkESVKA937QD5cIwOFLDFt0MQoXOEr9Q==
  /@types/node/11.13.17:
    dev: false
    resolution:
      integrity: sha512-7W3kSMa8diVH6s24a8Qrmvwu+vG3ahOC/flMHFdWSdnPYoQI0yPO84h5zOWYXAha2Npn3Pw3SSuQSwBUfaniyQ==
  /@types/ramda/0.26.15:
    dev: false
    resolution:
      integrity: sha512-kmIO8GVOLv0A8UxFr5lJaqp4g8jbXNhH0grBaPL6NgkOqSirRKpV9+5DTmIDXTyq2G18YMdxDhnScEZtY4brtQ==
  /fs-extra/8.1.0:
    dependencies:
      graceful-fs: 4.2.0
      jsonfile: 4.0.0
      universalify: 0.1.2
    dev: false
    engines:
      node: '>=6 <7 || >=8'
    resolution:
      integrity: sha512-yhlQgA6mnOJUKOsRUFsgJdQCvkKhcz8tlZG5HBQfReYZy46OwLcY+Zia0mtdHsOo9y/hP+CxMN0TU9QxoOtG4g==
  /graceful-fs/4.2.0:
    dev: false
    resolution:
      integrity: sha512-jpSvDPV4Cq/bgtpndIWbI5hmYxhQGHPC4d4cqBPb4DLniCfhJokdXhwhaDuLBGLQdvvRum/UiX6ECVIPvDXqdg==
  /jsonfile/4.0.0:
    dev: false
    optionalDependencies:
      graceful-fs: 4.2.0
    resolution:
      integrity: sha1-h3Gq4HmbZAdrdmQPygWPnBDjPss=
  /ramda/0.26.1:
    dev: false
    resolution:
      integrity: sha512-hLWjpy7EnsDBb0p+Z3B7rPi3GDeRG5ZtiI33kJhTt+ORCd38AbAIjB/9zRIUoeTbE/AVX5ZkU7m6bznsvrf8eQ==
  /tslib/1.10.0:
    dev: false
    resolution:
      integrity: sha512-qOebF53frne81cf0S9B41ByenJ3/IuH8yJKngAX35CmiZySA0khhkovshKK+jGCaMnVomla7gVlIcc3EvKPbTQ==
  /universalify/0.1.2:
    dev: false
    engines:
      node: '>= 4.0.0'
    resolution:
      integrity: sha512-rBJeI5CXAlmy1pV+617WB9J63U6XcazHHF2f2dbJix4XzpUF0RS3Zbj0FGIOCAva5P/d/GBOYaACQ1w+0azUkg==
specifiers:
  '@types/fs-extra': ^8.0.0
  '@types/node': ^11.12.0
  '@types/ramda': ^0.26.5
  fs-extra: ^8.0.0
  monorepo-shared-assets: ^0.0.17
  ramda: ^0.26.1
  tslib: ^1.9.3
  unique-temp-path: '*'
