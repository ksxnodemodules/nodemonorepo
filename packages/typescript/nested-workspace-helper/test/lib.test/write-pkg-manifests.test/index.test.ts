import * as fsx from 'fs-extra'
import * as xjest from 'extra-jest'
import * as fsTreeUtils from 'fs-tree-utils'
import {writePackageManifests} from '../../../index'
import createSetupTeardown from '../../.lib/setup-teardown'

const {apply} = createSetupTeardown('valid.yaml')
const snap = (x: any) => xjest.snap.safe(x)()
const snapFsTree = async () => snap(await fsTreeUtils.read.nested('root'))

const transformer: writePackageManifests.Transformer = item =>
  ({
    ...item.manifestContent,
    extra: {
      dirname: item.path,
      filename: item.manifestFile
    }
  })

it('works with default parameters', apply(async () => {
  await writePackageManifests('root', transformer)
  await snapFsTree()
}))

it('works with specified options', apply(async () => {
  await writePackageManifests(
    'root',
    transformer,
    {
      indentation: 4,
      finalNewLine: 0
    }
  )

  await snapFsTree()
}))
