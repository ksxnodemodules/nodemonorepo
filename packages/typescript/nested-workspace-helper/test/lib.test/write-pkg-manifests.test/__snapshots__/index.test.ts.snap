// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works with default parameters 1`] = `
"
node_modules:
  ignoreda: |
    {
      \\"name\\": \\"ignoreda\\",
      \\"version\\": \\"2.2.2\\"
    }
  ignoredb: |
    {
      \\"name\\": \\"ignoredb\\",
      \\"version\\": \\"3.3.3\\"
    }
package.json: |
  {
    \\"name\\": \\"root\\",
    \\"version\\": \\"0.0.0\\",
    \\"private\\": true,
    \\"workspaces\\": [
      \\"wrkspc0dept/*\\",
      \\"wrkspc1dept/*/*\\",
      \\"wrkspc2dept/*/*/*\\"
    ],
    \\"devDependencies\\": {
      \\"pkg00\\": \\"^0.1.0\\"
    },
    \\"extra\\": {
      \\"dirname\\": \\"root\\",
      \\"filename\\": \\"root/package.json\\"
    }
  }
wrkspc0dept:
  pkg00:
    package.json: |
      {
        \\"name\\": \\"pkg00\\",
        \\"version\\": \\"0.1.2\\",
        \\"extra\\": {
          \\"dirname\\": \\"root/wrkspc0dept/pkg00\\",
          \\"filename\\": \\"root/wrkspc0dept/pkg00/package.json\\"
        }
      }
  pkg01:
    package.json: |
      {
        \\"name\\": \\"pkg01\\",
        \\"version\\": \\"1.2.3\\",
        \\"private\\": false,
        \\"extra\\": {
          \\"dirname\\": \\"root/wrkspc0dept/pkg01\\",
          \\"filename\\": \\"root/wrkspc0dept/pkg01/package.json\\"
        }
      }
wrkspc1dept:
  sub10:
    pkg100:
      package.json: |
        {
          \\"name\\": \\"pkg100\\",
          \\"version\\": \\"2.2.5\\",
          \\"extra\\": {
            \\"dirname\\": \\"root/wrkspc1dept/sub10/pkg100\\",
            \\"filename\\": \\"root/wrkspc1dept/sub10/pkg100/package.json\\"
          }
        }
  sub11:
    pkg110:
      package.json: |
        {
          \\"name\\": \\"pkg110\\",
          \\"version\\": \\"22.3.5\\",
          \\"dependencies\\": {
            \\"pkg01\\": \\"^1.0.0\\",
            \\"pkg100\\": \\"~2.2.1\\"
          },
          \\"extra\\": {
            \\"dirname\\": \\"root/wrkspc1dept/sub11/pkg110\\",
            \\"filename\\": \\"root/wrkspc1dept/sub11/pkg110/package.json\\"
          }
        }
    pkg111prv:
      package.json: |
        {
          \\"name\\": \\"pkg111prv\\",
          \\"private\\": true,
          \\"version\\": \\"33.22.11\\",
          \\"dependencies\\": {
            \\"pkg01\\": \\"*\\"
          },
          \\"extra\\": {
            \\"dirname\\": \\"root/wrkspc1dept/sub11/pkg111prv\\",
            \\"filename\\": \\"root/wrkspc1dept/sub11/pkg111prv/package.json\\"
          }
        }
wrkspc2dept:
  sub20:
    sub200:
      pkg2000:
        node_modules:
          ignored2000a: |
            {
              \\"name\\": \\"ignored2000a\\",
              \\"version\\": \\"1.2.3\\"
            }
        package.json: |
          {
            \\"name\\": \\"pkg2000\\",
            \\"version\\": \\"123.456.789\\",
            \\"devDependencies\\": {
              \\"pkg111prv\\": \\"33.22.11\\",
              \\"foreign-foo\\": \\"^1.2.3\\"
            },
            \\"extra\\": {
              \\"dirname\\": \\"root/wrkspc2dept/sub20/sub200/pkg2000\\",
              \\"filename\\": \\"root/wrkspc2dept/sub20/sub200/pkg2000/package.json\\"
            }
          }

"
`;

exports[`works with specified options 1`] = `
"
node_modules:
  ignoreda: |
    {
      \\"name\\": \\"ignoreda\\",
      \\"version\\": \\"2.2.2\\"
    }
  ignoredb: |
    {
      \\"name\\": \\"ignoredb\\",
      \\"version\\": \\"3.3.3\\"
    }
package.json: |-
  {
      \\"name\\": \\"root\\",
      \\"version\\": \\"0.0.0\\",
      \\"private\\": true,
      \\"workspaces\\": [
          \\"wrkspc0dept/*\\",
          \\"wrkspc1dept/*/*\\",
          \\"wrkspc2dept/*/*/*\\"
      ],
      \\"devDependencies\\": {
          \\"pkg00\\": \\"^0.1.0\\"
      },
      \\"extra\\": {
          \\"dirname\\": \\"root\\",
          \\"filename\\": \\"root/package.json\\"
      }
  }
wrkspc0dept:
  pkg00:
    package.json: |-
      {
          \\"name\\": \\"pkg00\\",
          \\"version\\": \\"0.1.2\\",
          \\"extra\\": {
              \\"dirname\\": \\"root/wrkspc0dept/pkg00\\",
              \\"filename\\": \\"root/wrkspc0dept/pkg00/package.json\\"
          }
      }
  pkg01:
    package.json: |-
      {
          \\"name\\": \\"pkg01\\",
          \\"version\\": \\"1.2.3\\",
          \\"private\\": false,
          \\"extra\\": {
              \\"dirname\\": \\"root/wrkspc0dept/pkg01\\",
              \\"filename\\": \\"root/wrkspc0dept/pkg01/package.json\\"
          }
      }
wrkspc1dept:
  sub10:
    pkg100:
      package.json: |-
        {
            \\"name\\": \\"pkg100\\",
            \\"version\\": \\"2.2.5\\",
            \\"extra\\": {
                \\"dirname\\": \\"root/wrkspc1dept/sub10/pkg100\\",
                \\"filename\\": \\"root/wrkspc1dept/sub10/pkg100/package.json\\"
            }
        }
  sub11:
    pkg110:
      package.json: |-
        {
            \\"name\\": \\"pkg110\\",
            \\"version\\": \\"22.3.5\\",
            \\"dependencies\\": {
                \\"pkg01\\": \\"^1.0.0\\",
                \\"pkg100\\": \\"~2.2.1\\"
            },
            \\"extra\\": {
                \\"dirname\\": \\"root/wrkspc1dept/sub11/pkg110\\",
                \\"filename\\": \\"root/wrkspc1dept/sub11/pkg110/package.json\\"
            }
        }
    pkg111prv:
      package.json: |-
        {
            \\"name\\": \\"pkg111prv\\",
            \\"private\\": true,
            \\"version\\": \\"33.22.11\\",
            \\"dependencies\\": {
                \\"pkg01\\": \\"*\\"
            },
            \\"extra\\": {
                \\"dirname\\": \\"root/wrkspc1dept/sub11/pkg111prv\\",
                \\"filename\\": \\"root/wrkspc1dept/sub11/pkg111prv/package.json\\"
            }
        }
wrkspc2dept:
  sub20:
    sub200:
      pkg2000:
        node_modules:
          ignored2000a: |
            {
              \\"name\\": \\"ignored2000a\\",
              \\"version\\": \\"1.2.3\\"
            }
        package.json: |-
          {
              \\"name\\": \\"pkg2000\\",
              \\"version\\": \\"123.456.789\\",
              \\"devDependencies\\": {
                  \\"pkg111prv\\": \\"33.22.11\\",
                  \\"foreign-foo\\": \\"^1.2.3\\"
              },
              \\"extra\\": {
                  \\"dirname\\": \\"root/wrkspc2dept/sub20/sub200/pkg2000\\",
                  \\"filename\\": \\"root/wrkspc2dept/sub20/sub200/pkg2000/package.json\\"
              }
          }

"
`;
